src/terminal/parser/stateMachine.cpp

```diff
@@ -1952,3 +2016,16 @@ void StateMachine::_AccumulateTo(const wchar_t wch, size_t& value) noexcept
         value = MAX_PARAMETER_VALUE;
     }
 }
+
+// Routine Description:
+// - Determines if the engine is in "Variable Length String" state, which is a combination
+//   of all states that are expecting a string that has a undetermined length.
+//
+// Arguments:
+// - <none>
+// Return Value:
+// - True if it is. False if it isn't.
+const bool StateMachine::_IsVariableLengthStringState() const noexcept
+{
+    return _state == VTStates::OscString || _state == VTStates::DcsPassThrough || _state == VTStates::SosPmApcString;

```

Comment:
From the graph here https://vt100.net/emu/dec_ansi_parser, these are the only three cases. Not sure if there's more.